internal/controllers/machine/machine_controller_noderef.go

```diff
@@ -296,10 +297,17 @@ func (r *Reconciler) patchNode(ctx context.Context, remoteClient client.Client,
 	hasTaintChanges := taints.RemoveNodeTaint(newNode, clusterv1.NodeUninitializedTaint)
 
 	// Set Taint to a node in an old MachineSet and unset Taint from a node in a new MachineSet
-	isOutdated, err := shouldNodeHaveOutdatedTaint(ctx, r.Client, m)
+	isOutdated, notFound, err := shouldNodeHaveOutdatedTaint(ctx, r.Client, m)
 	if err != nil {
 		return errors.Wrapf(err, "failed to check if Node %s is outdated", klog.KRef("", node.Name))
 	}
+
+	// It is not possible to identify if we have to set the NodeOutdatedRevisionTaint if there is
+	// no OwnerReference or the owning MachineSet or MachineDeployment was not found.
+	// Note: This could happen e.g. during background deletion of objects.
+	if notFound {
```

Comment:
/hold

I think we're now skipping patches if hasAnnotationChanges or hasLabelChanges is true

I think we should only skip the if/else with isOutdated


(Please also check the test coverage to cover these cases correctly)

Reply:
added the test-case and adjusted the code+comment

Reply:
Yup good point. But yeah let's improve that as well :D